[Enhancement] CLion doesn't work after import the project

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Description

After import the project in CLion, some errors occur.

```shell
/Applications/CLion.app/Contents/bin/cmake/mac/bin/cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_MAKE_PROGRAM=/Applications/CLion.app/Contents/bin/ninja/mac/ninja -G Ninja -S /Users/adonis/Programs/doris/be -B /Users/adonis/Programs/doris/be/cmake-build-debug
-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- The C compiler identification is AppleClang 14.0.0.14000029
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- GLIBC_COMPATIBILITY is ON
-- Build type is DEBUG
-- Build target arch is arm64
-- make test: OFF
-- build gensrc if necessary
/Library/Developer/CommandLineTools/usr/bin/make -C script
/Users/adonis/Programs/doris/gensrc/script/gen_build_version.sh
/Library/Developer/CommandLineTools/usr/bin/make -C proto
make[1]: Nothing to be done for `all'.
/Library/Developer/CommandLineTools/usr/bin/make -C thrift
find: /installed/bin/thrift: No such file or directory
make[1]: Nothing to be done for `all'.
-- Could NOT find Boost: missing: system date_time (found /opt/homebrew/lib/cmake/Boost-1.80.0/BoostConfig.cmake (found suitable version "1.80.0", minimum required is "1.73.0"))
-- Could NOT find Boost: missing: system container (found /opt/homebrew/lib/cmake/Boost-1.80.0/BoostConfig.cmake (found suitable version "1.80.0", minimum required is "1.73.0"))
-- Looking for sched_getcpu
-- Looking for sched_getcpu - not found
-- Compiler Flags:  -Wall -Wno-sign-compare -pthread -Werror -fstrict-aliasing -fno-omit-frame-pointer -std=gnu++17 -D__STDC_FORMAT_MACROS -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DBOOST_SYSTEM_NO_DEPRECATED -DBRPC_ENABLE_CPU_PROFILER -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX=1 -Wno-attributes -DS2_USE_GFLAGS -DS2_USE_GLOG -DDORIS_WITH_MYSQL -faligned-new  -g -Wno-unused-local-typedefs -Og 
-- DORIS_DEPENDENCIES is backtrace;rocksdb;cyrus-sasl;libs2;snappy;Boost::date_time;thrift;thriftnb;glog;re2;hyperscan;pprof;lz4;libevent;libevent_pthreads;idn;gsasl;curl;libz;libbz2;gflags;brpc;protobuf;openssl;crypto;leveldb;bitshuffle;roaring;fmt;jemalloc;jemalloc_arrow;brotlicommon;brotlidec;brotlienc;zstd;arrow;parquet;orc;odbc;cctz;minizip;opentelemetry_common;opentelemetry_exporter_zipkin_trace;opentelemetry_resources;opentelemetry_version;opentelemetry_exporter_ostream_span;opentelemetry_trace;opentelemetry_http_client_curl;opentelemetry_exporter_otlp_http;opentelemetry_exporter_otlp_http_client;opentelemetry_otlp_recordable;opentelemetry_proto;aws-sdk-s3;aws-sdk-core;aws-sdk-transfer;aws-checksums;aws-c-io;aws-c-event-stream;aws-c-common;aws-c-cal;aws-c-s3;aws-c-auth;aws-crt-cpp;aws-c-compression;aws-c-http;aws-c-mqtt;aws-sdk-s3-crt;librdkafka_cpp;librdkafka;hdfs3;xml2;lzma;simdjson;Boost::container;bfd;iberty;intl;krb5support;krb5;com_err;gssapi_krb5;k5crypto;mysql
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Looking for sys/random.h
-- Looking for sys/random.h - found
-- Some symbols from glibc will be replaced for compatibility
-- Link Flags: Agent;Common;Env;Exec;Exprs;Gutil;IO;Olap;Rowset;Runtime;Service;Udf;Util;DorisGen;Webserver;Geo;Vec;backtrace;rocksdb;cyrus-sasl;libs2;snappy;Boost::date_time;thrift;thriftnb;glog;re2;hyperscan;pprof;lz4;libevent;libevent_pthreads;idn;gsasl;curl;libz;libbz2;gflags;brpc;protobuf;openssl;crypto;leveldb;bitshuffle;roaring;fmt;jemalloc;jemalloc_arrow;brotlicommon;brotlidec;brotlienc;zstd;arrow;parquet;orc;odbc;cctz;minizip;opentelemetry_common;opentelemetry_exporter_zipkin_trace;opentelemetry_resources;opentelemetry_version;opentelemetry_exporter_ostream_span;opentelemetry_trace;opentelemetry_http_client_curl;opentelemetry_exporter_otlp_http;opentelemetry_exporter_otlp_http_client;opentelemetry_otlp_recordable;opentelemetry_proto;aws-sdk-s3;aws-sdk-core;aws-sdk-transfer;aws-checksums;aws-c-io;aws-c-event-stream;aws-c-common;aws-c-cal;aws-c-s3;aws-c-auth;aws-crt-cpp;aws-c-compression;aws-c-http;aws-c-mqtt;aws-sdk-s3-crt;librdkafka_cpp;librdkafka;hdfs3;xml2;lzma;simdjson;Boost::container;bfd;iberty;intl;krb5support;krb5;com_err;gssapi_krb5;k5crypto;mysql;-lapple_nghttp2;-lresolv;-liconv;tcmalloc;glibc-compatibility-explicit;glibc-compatibility;-framework CoreFoundation;-framework CoreGraphics;-framework CoreText;-framework Foundation;-framework SystemConfiguration;-framework Security
-- Performing Test HAS_WARNING_OPTION
-- Performing Test HAS_WARNING_OPTION - Success
-- dir='/Users/adonis/Programs/doris/be/src'
-- dir='/Users/adonis/Programs/doris/be/test'
-- dir='/Users/adonis/Programs/doris/be/../gensrc/build'
-- dir='/installed/include'
-- dir='/installed/gperftools/include'
-- dir='/Users/adonis/.jenv/versions/openjdk64-11.0.16.1/include'
-- dir='/Users/adonis/.jenv/versions/openjdk64-11.0.16.1/include/darwin'
-- Configuring done
CMake Error at src/service/CMakeLists.txt:41 (target_link_libraries):
  Target "doris_be" links to:

    Boost::date_time

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

### Solution

The environment variable `DORIS_THIRDPARTY` and some options must be specified by default in `CMakeLists.txt`.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/be/CMakeLists.txt
+++ b/be/CMakeLists.txt
@@ -65,8 +65,22 @@ else()
     add_definitions(-DBOOST_STACKTRACE_USE_NOOP)
 endif()
 
-option(GLIBC_COMPATIBILITY "Enable compatibility with older glibc libraries." ON)
+# Options
+if (NOT OS_MACOSX)
+    option(GLIBC_COMPATIBILITY "Enable compatibility with older glibc libraries." ON)
+    option(USE_LIBCPP "Use libc++" OFF)
+    option(USE_MEM_TRACKER, "Use memory tracker" ON)
+else()
+    option(GLIBC_COMPATIBILITY "Enable compatibility with older glibc libraries." OFF)
+    option(USE_LIBCPP "Use libc++" ON)
+    option(USE_MEM_TRACKER, "Use memory tracker" OFF)
+endif()
+option(USE_JEMALLOC "Use jemalloc" ON)
+
 message(STATUS "GLIBC_COMPATIBILITY is ${GLIBC_COMPATIBILITY}")
+message(STATUS "USE_LIBCPP is ${USE_LIBCPP}")
+message(STATUS "USE_MEM_TRACKER is ${USE_MEM_TRACKER}")
+message(STATUS "USE_JEMALLOC is ${USE_JEMALLOC}")
 
 # set CMAKE_BUILD_TYPE
 if (NOT CMAKE_BUILD_TYPE)
@@ -86,13 +100,17 @@ message(STATUS "Build target arch is ${CMAKE_BUILD_TARGET_ARCH}")
 
 # Set dirs
 set(BASE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
-set(ENV{DORIS_HOME} "${BASE_DIR}/../")
+set(ENV{DORIS_HOME} "${BASE_DIR}/..")
 set(BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}")
-set(THIRDPARTY_DIR "$ENV{DORIS_THIRDPARTY}/installed/")
 set(GENSRC_DIR "${BASE_DIR}/../gensrc/build/")
 set(SRC_DIR "${BASE_DIR}/src/")
 set(TEST_DIR "${CMAKE_SOURCE_DIR}/test/")
 set(OUTPUT_DIR "${BASE_DIR}/output")
+if (NOT DEFINED ENV{DORIS_THIRDPARTY})
+    set(ENV{DORIS_THIRDPARTY} "$ENV{DORIS_HOME}/thirdparty")
+endif()
+set(THIRDPARTY_DIR "$ENV{DORIS_THIRDPARTY}/installed")
+message(STATUS "THIRDPARTY_DIR is ${THIRDPARTY_DIR}")
 
 option(MAKE_TEST "ON for make unit test or OFF for not" OFF)
 message(STATUS "make test: ${MAKE_TEST}")

--- a/be/src/tools/CMakeLists.txt
+++ b/be/src/tools/CMakeLists.txt
@@ -35,9 +35,11 @@ target_link_libraries(meta_tool
 install(DIRECTORY DESTINATION ${OUTPUT_DIR}/lib/)
 install(TARGETS meta_tool DESTINATION ${OUTPUT_DIR}/lib/)
 
+if (NOT OS_MACOSX)
 # Meta tool never need debug info
 add_custom_command(TARGET meta_tool POST_BUILD
     COMMAND ${CMAKE_OBJCOPY} --only-keep-debug $<TARGET_FILE:meta_tool> $<TARGET_FILE:meta_tool>.dbg
     COMMAND ${CMAKE_STRIP} --strip-debug --strip-unneeded $<TARGET_FILE:meta_tool>
     COMMAND ${CMAKE_OBJCOPY} --add-gnu-debuglink=$<TARGET_FILE:meta_tool>.dbg $<TARGET_FILE:meta_tool>
     )
+endif()

